src/foundation/src/PDFsharp/src/PdfSharp/Pdf.Signatures/PdfSignatureHandler.cs

```diff
@@ -110,26 +110,31 @@ private void ComputeSignatureAndRange(object sender, PdfDocumentEventArgs e)
 
         private void AddSignatureComponents(object sender, EventArgs e)
         {
+            if (Options.PageIndex < 0 || Options.PageIndex >= Document.PageCount)
```

Comment:
I'd suggest to check for 'Options.PageIndex < 0' in constructor (don't don unnecessary work if we know it will crash) and throw an ArgumentException or ArgumentOutOfRangeException in constructor if < 0. Don't forget to handle nullity with 'Options?.PageIndex < 0'